src/daos-deserialize/daos-deserialize.c

```diff
@@ -26,12 +24,107 @@ void print_usage(void)
 }
 
+/* if a directory is passed in we need to count the files in it,
+ * and read them into the paths array */
+static int count_files(char **argpaths, char ***paths, int *numpaths) {
+    int         rc = 0;
+    int         i;
+    struct stat statbuf;
+
+    rc = stat(argpaths[0], &statbuf);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to stat input file"
+                MFU_ERRF, MFU_ERRP(-MFU_ERR_INVAL_ARG));
+        rc = 1;
+        goto out;
+    }
+
+    if (S_ISDIR(statbuf.st_mode)) {
+        DIR *dir;
+        struct dirent *entry;
+        int num_files = 0;
+
+        /* set a max paths so that we do not 
+         * have to read directory twice,
+         * once to get number of files, and
+         * once to copy/save strings to paths array */
+        int max_paths = 1024;
+
+        /* use paths instead of argpaths in case directory is used */
+        *paths = MFU_MALLOC(max_paths * sizeof(char*));
+        if (*paths == NULL) {
+            rc = ENOMEM;
+            goto out;
+        }
+
+        dir = opendir(argpaths[0]);
+        if (dir == NULL) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to open directory"
+                    MFU_ERRF, MFU_ERRP(-MFU_ERR_INVAL_ARG));
+            rc = 1;
+            goto out;
+        }
+
+        struct stat stbuf;
+        while((entry = readdir(dir)) != NULL) {
+            rc = stat(entry->d_name, &stbuf);
+            if (S_ISREG(stbuf.st_mode)) {   
+                char *path = MFU_STRDUP(entry->d_name);
+                if (path == NULL) {
+                    rc = ENOMEM;
+                    goto out;
+                }
+                int path_len = strlen(argpaths[0]) + strlen(path) + 2;
+                int len = 0;
+                (*paths)[num_files] = MFU_MALLOC(path_len);
+                if ((*paths)[num_files] == NULL) {
+                    MFU_LOG(MFU_LOG_ERR, "failed to allocate path");
+                    rc = ENOMEM;
+                    goto out;
+                }
+                len = snprintf((*paths)[num_files], path_len, "%s/%s", argpaths[0],
+                               path);
+                if (len >= path_len) {
+                    MFU_LOG(MFU_LOG_ERR, "filename is too long");
+                    rc = 1;
+                    mfu_free(&path);
+                    goto out;
+                }
+                mfu_free(&path);
+                num_files++;
+            } else {
+                continue;
+            }
+        }
+        *numpaths = num_files;
+        closedir(dir);
+    } else {
+        /* use paths instead of argptahs in case directory is used */
+        *paths = MFU_MALLOC(*numpaths * sizeof(char*));
+        if (*paths == NULL) {
+            rc = ENOMEM;
+            goto out;
+        }
+
+        for (i = 0; i < *numpaths; i++) {
+            (*paths)[i] = MFU_STRDUP(argpaths[i]);
+            if ((*paths)[i] == NULL) {
+                rc = ENOMEM;
+                goto out;
+            }
```

Comment:
@dsikich similar to above
```suggestion
            if ((*paths)[i] == NULL) {
                MFU_LOG(MFU_LOG_ERR, "Failed to allocate paths.");
                rc = 1;
                goto out;
            }
```